ai/STA/Tactic/position_for_pass.py

```diff
@@ -39,61 +42,63 @@ def __init__(self, game_state: GameState, player: OurPlayer, target: Pose=Pose()
-        return Pose(self.target.position, destination_orientation)
+        self.game_state.debug_interface.add_point(self.target_position, COLOR_ID_MAP[4], width=5,
+                                                  timeout=0.1)
+        return Pose(self.target_position, destination_orientation)
 
     def _find_best_player_position(self):
         if self.auto_position:
             pad = 200
-            if self.player.id == 1: #role is 'top_defence':
-                A = Position(GameState().const["FIELD_OUR_GOAL_X_EXTERNAL"]+pad, GameState().const["FIELD_Y_TOP"]-pad)
-                B = Position(0-pad, (GameState().const["FIELD_Y_TOP"] / 3)+pad)
-                self.target.position = best_position_in_region(self.player, A, B)
-            elif self.player.id == 2: #player.role is 'bottom_defence':
-                A = Position(GameState().const["FIELD_OUR_GOAL_X_EXTERNAL"]+pad, GameState().const["FIELD_Y_BOTTOM"]+pad)
-                B = Position(0-pad, (GameState().const["FIELD_Y_BOTTOM"] / 3)-pad)
-                self.target.position = best_position_in_region(self.player, A, B)
-            elif self.player.id == 3: #player.role is 'top_offence':
-                A = Position(GameState().const["FIELD_THEIR_GOAL_X_EXTERNAL"]-pad, GameState().const["FIELD_Y_TOP"]-pad)
-                B = Position(0+pad, 0+pad)
-                self.target.position = best_position_in_region(self.player, A, B)
-            elif self.player.id == 4: #player.role is 'bottom_offence':
-                A = Position(GameState().const["FIELD_THEIR_GOAL_X_EXTERNAL"]-pad, GameState().const["FIELD_Y_BOTTOM"]+pad)
-                B = Position(0+pad, 0-pad)
-                self.target.position = best_position_in_region(self.player, A, B)
-            elif self.player.id == 5: #player.role is 'center':
-                A = Position(GameState().const["FIELD_OUR_GOAL_X_EXTERNAL"]-pad, (GameState().const["FIELD_Y_BOTTOM"] / 3)-pad)
-                B = Position(0-pad, GameState().const["FIELD_Y_TOP"] / 3)
-                self.target.position = best_position_in_region(self.player, A, B)
\ No newline at end of file
+            if GameState().const["FIELD_OUR_GOAL_X_EXTERNAL"] > 0:
+                our_goal_field_limit = GameState().const["FIELD_OUR_GOAL_X_EXTERNAL"] - pad
+                our_side_center_field_limit = pad
+                their_goal_field_limit = GameState().const["FIELD_THEIR_GOAL_X_EXTERNAL"] + pad
+                their_side_center_field_limit = -pad
+            else:
+                our_goal_field_limit = GameState().const["FIELD_OUR_GOAL_X_EXTERNAL"] + pad
+                our_side_center_field_limit = -pad
+                their_goal_field_limit = GameState().const["FIELD_THEIR_GOAL_X_EXTERNAL"] - pad
+                their_side_center_field_limit = pad
+
+            role = GameState().get_role_by_player_id(self.player.id)
+            if role is Role.FIRST_DEFENCE:  # role is 'top_defence':
```

Comment:
Il serait plus propre de faire un dictionnaire et une for loop pour évité la répétition de code, puisque les lignes suivantes sont tous pareils uniquement le Role et la position A et B change.